File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    /*
        PASCAL TRIANGLE:
        if the index is first or last -> value will be 1
        else value will be -> [row-1][col-1] + [row-1][col]
    */
    public List<List<Integer>> generate(int numRows) {
        // output list
        List<List<Integer>> output = new ArrayList<>();
        for(int i = 0;i<numRows;i++){
            // generating current row elements
            List<Integer> currentRow = new ArrayList<>();
            for(int j =0;j<=i;j++){
                if(j == 0 || j == i)currentRow.add(1);
                else currentRow.add(output.get(i-1).get(j-1)+output.get(i-1).get(j));
            }
            output.add(currentRow);
        }
        return output;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    #
    #   PASCAL TRIANGLE:
    #   if the index is first or last -> value will be 1
    #   else value will be -> [row-1][col-1] + [row-1][col]
    #
    
    def generate(self, numRows: int) -> List[List[int]]:
        output = []
        for i in range(numRows):
            currentRow = [0]*(i+1)
            for j in range(i+1):
                if j == 0 or j == i:
                    currentRow[j]=1
                else:
                    currentRow[j] = output[i-1][j-1] + output[i-1][j]
            output.append(currentRow)
        return output